monai/metrics/cumulative_average.py

```diff
@@ -9,64 +9,170 @@
+        """
+        if self.is_distributed:
+            if avg:
+                x = x / dist.get_world_size(group=self.group)
+            else:
+                x = x.clone()
+            dist.all_reduce(x, group=self.group)
+
+        return x
+
+    def get_current(self, to_numpy: bool = True) -> Union[NDArray, torch.Tensor]:
+        """
+        returns the most recent value (averaged across processes)
+
+        Args:
+            to_numpy: whether to convert to numpy array. Defaults to True
+        """
+        x = self.reduce(self.val, avg=True)
+        if to_numpy:
+            x = x.cpu().numpy()
+        return x
+
+    def aggregate(self, to_numpy: bool = True) -> Union[NDArray, torch.Tensor]:
+        """
+        returns the total average value (averaged across processes)
+
+        Args:
+            to_numpy: whether to convert to numpy array. Defaults to True
+        """
+        sum = self.reduce(self.sum)
+        count = self.reduce(self.count)
+
+        x = torch.where(count > 0, sum / count, sum)
+        if to_numpy:
+            x = x.cpu().numpy()
+        return x
+
+    def append(self, val: Any, count: Any = 1) -> None:
+        """
+        Append with a new value, and an optional count
+
+        Args:
+            val: new value (e.g. constant, list, numpy array or Tensor) to keep track of.
+            count: new count (e.g. constant, list, numpy array or Tensor), to update the contribution count
+
+        For example:
+            # a simple constant tracking
+            avg = CumulativeAverage()
+            avg.append(0.6)
+            avg.append(0.8)
+            print(avg.aggregate()) #prints 0.7
+
+            # an array tracking, e.g. metrics from 3 classes
+            avg= CumulativeAverage()
+            avg.append([0.2, 0.4, 0.4])
+            avg.append([0.4, 0.6, 0.4])
+            print(avg.aggregate()) #prints [0.3, 0.5. 0.4]
+
+            # different contributions / counts
+            avg= CumulativeAverage()
+            avg.append(1, count=4) #avg metric 1 coming from a batch of 4
+            avg.append(2, count=6) #avg metric 2 coming from a batch of 6
+            print(avg.aggregate()) #prints 1.6 == (1*4 +2*6)/(4+6)
+
+            # different contributions / counts
+            avg= CumulativeAverage()
+            avg.append([0.5, 0.5, 0], count=[1, 1, 0]) # last elements count is zero to ignore it
+            avg.append([0.5, 0.5, 0.5], count=[1, 1, 1]) #
+            print(avg.aggregate()) #prints [0.5, 0.5, 0,5] == ([0.5, 0.5, 0] + [0.5, 0.5, 0.5]) / ([1, 1, 0] + [1, 1, 1])
+
+        """
 
-        return self.sum / self.not_nans
+        val = self.proper_tensor(val)
+        count = self.proper_tensor(count)
+
+        val_count = val * count
+        nfin = ~torch.isfinite(val_count)
+        if torch.any(nfin):
+            # non-finite numbers may indicate some errors in the user code
+            print("non-finite numbers received", val, count)
```

Comment:
We shouldn't have print statements in regular code since at times we would want programs to be silent. Instead if it's important to state this we should write to the log or issue a warning with `warnings.warn`.

Reply:
okay, I 've changed it to warnings. thank you